src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -2254,6 +2254,7 @@ class AlterTableStatementSegment(BaseSegment):
                     Ref.keyword("COLUMN", optional=True),
                     Ref("ColumnDefinitionSegment"),
                 ),
+                Sequence("DROP", Ref.keyword("COLUMN"), Ref("ColumnReferenceSegment")),
```

Comment:
Should we add optional `IF EXISTS`?
https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-table-transact-sql?view=sql-server-ver16